### ISSUE TITLE
Incorrect prior_association() ? 

### ISSUE THREAD

**Issue by sappersapper** [spencerbraun/anomaly_transformer_pytorch#5]
It seems the method prior_association() can not back propagate gradient to train Ws, or maybe I misunderstood?
https://github.com/spencerbraun/anomaly_transformer_pytorch/blob/6d15200911260eee910a3664d70f07886c47708b/model.py#L41-L45

according to paper, is this the right way?
`gaussian = 1 / math.sqrt(2 * math.pi) / self.sigma * torch.exp(- 0.5 * (p / self.sigma).pow(2))`

**Comment by hancan0** [comment]
Yes, I found the same problem.  gaussian = 1 / math.sqrt(2 * math.pi) / self.sigma * torch.exp(- 0.5 * (p / self.sigma).pow(2)).And I emailed the author about it.

**Comment by sappersapper** [comment]
Does the `self.sigma` need to be constrained positive? How to constrain it?

**Comment by DanyaXu** [comment]
> Yes, I found the same problem. gaussian = 1 / math.sqrt(2 * math.pi) / self.sigma * torch.exp(- 0.5 * (p / self.sigma).pow(2)).And I emailed the author about it.

I also encountered the same problem, did the author reply to you?

**Comment by hancan0** [comment]
> > Yes, I found the same problem. gaussian = 1 / math.sqrt(2 * math.pi) / self.sigma * torch.exp(- 0.5 * (p / self.sigma).pow(2)).And I emailed the author about it.
> 
> I also encountered the same problem, did the author reply to you?

Yes, The author agrees it.
